test/TestImport.hs

```diff
@@ -24,4 +27,28 @@ withApp = before $ do
         ["config/test-settings.yml", "config/settings.yml"]
         []
         ignoreEnv
-    makeFoundation settings
+    foundation <- makeFoundation settings
+    wipeDB foundation
+    return foundation
+
+-- This function will truncate all of the tables in your database.
+-- 'withApp' calls it before each test, creating a clean environment for each
+-- spec to run in.
+wipeDB :: App -> IO ()
+wipeDB app = do
+    runDBWithApp app $ do
+        tables <- getTables
+        let queries = map ("TRUNCATE TABLE " ++ ) tables
+
+        -- In MySQL, a table cannot be truncated if another table references it via foreign key.
+        -- Since we're wiping both the parent and child tables, though, it's safe
+        -- to temporarily disable this check.
+        rawExecute "SET foreign_key_checks = 0;" []
+        forM_ queries (\q -> rawExecute q [])
+        rawExecute "SET foreign_key_checks = 1;" []
+    return ()
+
+getTables :: MonadIO m => ReaderT SqlBackend m [Text]
+getTables = do
+    tables <- rawSql "SHOW TABLES;" []
```

Comment:
So I'd be super concerned in a production setting, but for a test environment I think its unlikely we'd do real damage.

The major case I can think of where you want a table not wiped is when you use a migration tool that maintains a table like `schema_migrations`. For our rails codebase at work, we have Jenkins run the migrations as a (partial) way of checking if they'll fail before we run them against production. For that to work, I'm pretty sure it needs `schema_migrations` to not be wiped. (this doesn't apply to a clean-slate environment like Travis CI). I'll look into this, but I suspect there isn't a general solution besides developers blacklisting tables themselves.

One disadvantage to not wiping all the tables is that it might leave the database in an inconsistent state with respect to referential integrity. If you have table A that persistent knows about, table B that it doesn't, and table B has a foreign key to table A, then wiping table A leaves the database inconsistent. For a database like Postgres, it'll raise an error if table B isn't in the list of tables passed to `TRUNCATE`; for MySQL I'd have to check if there's a decent solution. This seems edge-casey though.

Anyway I need to start work, will come back to this later tonight.